src/js/utils/JSONUtil.js

```diff
@@ -0,0 +1,299 @@
+/**
+ * Tokenizer regular expressions
+ */
+const TOKENIZER_TOKENS = {
+  'comma': /,/g,
+  'end-label': /:/g,
+  'begin-object': /\{/g,
+  'end-object': /\}/g,
+  'begin-array': /\[/g,
+  'end-array': /\]/g,
+  'string': /"(\\["\\/bfnrtu"]|[^"\\"])*"/g,
+  'maybe-string': /"([^"]|\\")*/g,
+  'null': /null/g,
+  'boolean': /(true|false)/g,
+  'number': /-?\d+(\.\d+)?([eE]-?\d+)?/g,
+  'maybe-decimal-number': /-?\d+\./g,
+  'maybe-negative-number': /-/g,
+  'maybe-exponential-number': /-?\d+(\.\d+)?([eE])?/g,
+  'maybe-exponential-number-negative': /-?\d+(\.\d+)?([eE]-)?/g,
+  'symbol': /\w+/g
+};
+
+// Extract only keys
+const TOKENIZER_TYPES = Object.keys( TOKENIZER_TOKENS );
+
+/**
+ * Return the next JSON token from the given string
+ *
+ * @param {string} chunk - The source payload
+ * @param {number} offset - The offset to start searching from
+ * @returns {array} - Returns an array with [ tokenType, tokenOffset ]
+ */
+function nextJSONToken( chunk, offset ) {
+  return TOKENIZER_TYPES.reduce(function (lastMatch, tokenName) {
+    let tokenRx = TOKENIZER_TOKENS[tokenName];
+
+    // Search for a token at given offset
+    tokenRx.lastIndex = offset;
+    let match = tokenRx.exec(chunk);
+
+    // If we got something closer than the previous match, use that one
+    if (match && match.index < lastMatch[2]) {
+      return [tokenName, match[0], match.index];
+    }
+
+    // Otherwise default
+    return lastMatch;
+
+  }, ['', '', chunk.length]);
+}
+
+/**
+ * Count how many lines exists in the string between
+ * given start and end indices.
+ *
+ * @param {string} src - The string to test
+ * @param {number} startAt - The index to start searching for new lines (inclusive)
+ * @param {number} endAt - The index to end searching for new lines (exclusive)
+ * @returns {number} - The number of new line characters found
+ */
+function countNewLines(src, startAt, endAt) {
+  let newLines = 0;
+  for (let i=startAt; i<endAt; i++) {
+    if (src[i] === '\n') {
```

Comment:
Yup, since windows use `\r\n` (so we have an extraneous `\r`, accounted as any other character in the string)